src/toolkit/pyroprocessing/pyre_refining.h

```diff
@@ -0,0 +1,139 @@
+  double Bisector(boost::function<double(double)> SolveFunc, double lower, double upper);
+
+  /// @brief a boost binded version of the Thermal function whose only parameter is temperature
+  boost::function<double(double)> ThermalFunc;
+
+  /// @brief Calculates the overall efficiency of the process by combining thermal, pressure and stirrer effs.
+  /// @return a value between 0 and 1 to be multiplied with separation efficiencies
+  double Efficiency();
+
+  /// @brief A function describing the Thermal Efficiency of the process
+  /// @param temp The operating temperature of the process
+  /// @param newEff the new efficiency due to diversion, used to solve for new temperature
+  /// @return a value between 0 and 1 to be used in the Efficiency function
+  double Thermal(double temp, double newEff);
+
+  /// @brief overloaded version of Thermal that uses getters for all parameters
+  double Thermal();
+
+  /// @brief A function describing the Pressure Efficiency of the process
+  /// @param prs The operating pressure of the electrorefiner
+  /// @param newEff the new efficiency due to diversion, used to solve for new pressure
+  /// @return a value between 0 and 1 to be used in the Efficiency function
+  double PressureEff(double prs, double newEff);
+
+  /// @brief overloaded version of PressureEff that uses getters for all parameters
+  double PressureEff();
+
+  /// @brief A function describing the Central Stirrer Efficiency of the process
+  /// @param rot The operating speed of the central stirrer
+  /// @param newEff the new efficiency due to diversion, used to solve for new rotation speed
+  /// @return a value between 0 and 1 to be used in the Efficiency function
+  double Agitation(double rot, double newEff);
+
+  /// @brief overloaded version of Agitation that uses getters for all parameters
+  double Agitation();
+
+  /// @brief Determines the throughput of the process based on facility parameters
+  /// @return material throughput kg / day
+  double Throughput();
+
+  /// @return the value for thermal efficiency coefficient 0
+  double t0();
+
+  /// @return the value for thermal efficiency coefficient 1
+  double t1();
+
+  /// @return the value for thermal efficiency coefficient 2
+  double t2();
+
+  /// @return the value for thermal efficiency coefficient 3
+  double t3();
+
+  /// @return the value for pressure efficiency coefficient 0
+  double p0();
+
+  /// @return the value for pressure efficiency coefficient 1
+  double p1();
+
+  /// @return the value for pressure efficiency coefficient 2
+  double p2();
+
+  /// @return the value for pressure efficiency coefficient 3
+  double p3();
+
+  /// @return the value for rotation efficiency coefficient 0
+  double a0();
+
+  /// @return the value for rotation efficiency coefficient 1
+  double a1();
+
+  /// @return the value for rotation efficiency coefficient 2
+  double a2();
+
+  /// @return the value for rotation efficiency coefficient 3
+  double a3();
```

Comment:
I don't understand where those methods are implemented..

Reply:
These are deprecated since changing to vectors, I've removed them